rust/src/longest_common_prefix.rs

```diff
@@ -0,0 +1,24 @@
+use crate::Solution;
+
```

Comment:
If you have any comments here you are welcome

Reply:
Didn't really like this approach by the first look with having vector of mutable char iterators and fetching them every by one every step. Mostly because the code becomes very cryptic for a reasonably simple algorithm. On the second look I've realized it's not that simple to come up with anything better (especially when I understood that my first idea of treating everything as byte-slices with `as_bytes` doesn't work for Unicode for example). So the only thing I could suggest instead is to convert everything to Vector<char>  like this:

```
let first: Vec<char> = strs[0].chars().collect();
        let vectors: Vec<Vec<char>> = strs[1..].iter().map(|s| s.chars().collect()).collect();

        for i in 0..first.len() {
            for s in &vectors {
                if s[i] != first[i] {
                    return first[..i].iter().collect();
                }
            }
        }
```

But this could be a significant performance pessimisation in cases when we have a lot of long strings and short common prefix for example. So I could not come up with anything better then your solution after all :)